core/dive.c

```diff
@@ -1208,13 +1208,26 @@ static void fixup_no_o2sensors(struct divecomputer *dc)
 	}
 }
 
+/* Remove sensor and gas pressure info */
+static void clear_sensor(struct sample *samples, int sample_id, int sensor_id)
+{
+	struct sample *s = &samples[sample_id];
+	if (sample_id == 0) {
+		s->sensor[sensor_id] = 0;
+		s->pressure[sensor_id].mbar = 0;
+	} else {
+		s->sensor[sensor_id] = s[-1].sensor[sensor_id];
+		s->pressure[sensor_id].mbar = s[-1].pressure[sensor_id].mbar;
```

Comment:
I don't think this code is sensible.

Yeah, yeah, I realize it was copied from previous code, but that previous code wasn't really sensible either, and this change actually expands upon the bad behavior.

In particular, let's say that we have a sample with an invalid sensor.  The old code in fixup_dc_sample_sensors() would mark that as having NO_SENSOR, which admittedly didn't work so well, but at least it was somewhat sensible, and it didn't change the pressure.

The *new* code now overwrites both the invalid sensor data _and_ the pressure data. The latter might have been valid, just with the wrong cylinder index. Hmm?